python/seldon_deploy_sdk/api_client.py

```diff
@@ -99,6 +100,31 @@ def user_agent(self, value):
     def set_default_header(self, header_name, header_value):
         self.default_headers[header_name] = header_value
 
+    def __call_api_with_retry(
+            self, resource_path, method, path_params=None,
+            query_params=None, header_params=None, body=None, post_params=None,
+            files=None, response_type=None, auth_settings=None,
+            _return_http_data_only=None, collection_formats=None,
+            _preload_content=True, _request_timeout=None):
+        try:
+            return self.__call_api(resource_path=resource_path,method=method,path_params=path_params,query_params=query_params,
+                            header_params=header_params,body=body,post_params=post_params,files=files,
+                            response_type=response_type, auth_settings=auth_settings,
+                            _return_http_data_only=_return_http_data_only, collection_formats=collection_formats,
+                            _preload_content=_preload_content,_request_timeout=_request_timeout)
+        except seldon_deploy_sdk.rest.ApiException as e:
+            #if unauthenticated and have authenticator try refreshing in case token expired
+            if e.status == 401 and self.authenticator is not None:
+                token = self.authenticator.authenticate()
```

Comment:
Should really be a refresh func and in the password grant implementation refresh just calls authenticate again. Or refresh strategy could be a config option. Might be future situations where refresh token could be desirable.